pulsar-common/src/main/java/org/apache/pulsar/common/nar/NarClassLoader.java

```diff
@@ -139,7 +139,7 @@ public class NarClassLoader extends URLClassLoader {
 
     private static final String TMP_DIR_PREFIX = "pulsar-nar";
 
-    public static final String DEFAULT_NAR_EXTRACTION_DIR = System.getProperty("java.io.tmpdir");
+    public static final String DEFAULT_NAR_EXTRACTION_DIR = System.getProperty("nar.extraction.tmpdir");
```

Comment:
You have to fallback on the legacy name if this property is not configured

Reply:
Has changed, if `nar.extraction.tmpdir` not configured will use the old config` java.io.tmpdir`.